lucene/lucene-directory/src/main/java/org/infinispan/lucene/FileMetadata.java

```diff
@@ -12,18 +13,26 @@
  */
 public final class FileMetadata {
 
-   @ProtoField(number = 1, defaultValue = "0")
-   int bufferSize;
+   private final int bufferSize;
 
-   @ProtoField(number = 2, defaultValue = "0")
-   long size = 0;
+   private long size;
 
-   FileMetadata() {}
+   @ProtoFactory
+   public FileMetadata(int bufferSize, long size) {
+      this.bufferSize = bufferSize;
+      this.size = size;
+   }
 
    public FileMetadata(int bufferSize) {
-      this.bufferSize = bufferSize;
+      this(bufferSize, 0);
+   }
+
+   @ProtoField(number = 1, defaultValue = "1024")
```

Comment:
Do we now need the `@ProtoField` annotations on the public getter methods because the global var is private?

Reply:
Yes. Private fields cannot be annotated because they are not readable. Final is fine. An unwritable/final field can be annotated if ProtoFactory exists (otherwise a setter is needed or the field should be non-final).